### PR TITLE
Generate `compatibility_date` in `wrangler init` and warn if it's missing

### PR DESCRIPTION
**Pull request by kentonv**
We'd like everyone to specify a `compatibility_date` in their `wrangler.toml` going forward. This change both adds the date automatically to newly-generated `wrangler.toml` files, and warns when parsing old ones that are missing it.

See also the documentation change: https://github.com/cloudflare/cloudflare-docs/pull/2289

### CODE REVIEW

**Comment by WalshyDev** [comment]
Ah great! I noticed this a few days ago and had it on my todo to report

I also notice the init does double quotes for the filled out values and single for the empty ones. Not sure if this is an issue, just an odd inconsistency

File: src/settings/toml/manifest.rs [hunk 1]
```diff
@@ -93,6 +94,13 @@ impl Manifest {
 
         check_for_duplicate_names(&manifest)?;
 
+        if manifest.compatibility_date.is_none() {
+            StdOut::warn(&format!(
+                "Your configuration file is missing compatibility_date, so a distant past date is assumed. To get the latest possibly-breaking bug fixes, add this line to your wrangler.toml:\n\n    compatibility_date = \"{}\"\n",
+                Utc::now().format("%F")));
+            StdOut::warn("For more information about compatibility dates, see: https://developers.cloudflare.com/workers/platform/compatibility-dates");
```
Comment (jyn514) [inline src/settings/toml/manifest.rs:101]: This URL doesn't seem to exist yet? Is that being worked on in parallel?

**Review by jyn514 — COMMENTED** [review]
Code looks fine, but @Electroid or @kabirsikand might want to take a look at the wording.

**Comment by jyn514** [comment]
I tested this locally - the warning looks like
```
⚠️  Your configuration file is missing compatibility_date, so a distant past date is assumed. To get the latest possibly-breaking bug fixes, add this line to your wrangler.toml:

    compatibility_date = "2021-09-14"

⚠️  For more information about compatibility dates, see: https://developers.cloudflare.com/workers/platform/compatibility-dates
```
and is shown whenever you run wrangler (I think being annoying is intentional).

File: src/settings/toml/manifest.rs (hunk 1 above)
Comment (WalshyDev) [reply src/settings/toml/manifest.rs:101]: Sorry not Kenton but yes - https://github.com/cloudflare/cloudflare-docs/pull/2289

**Comment by Electroid** [comment]
Wording is good with me. ✅

**Review by Electroid — APPROVED** [review]

**Pull request merged**